oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java

```diff
@@ -1263,6 +1275,11 @@ private void combineInternalPropRemovals(final NodeDocument doc, final UpdateOp
         private boolean isDeletedOrOrphanedNode(final NodeState traversedState, final Path greatestExistingAncestorOrSelf,
                                                 final GCPhases phases, final NodeDocument doc) {
 
+            if (fullGcMode == EMPTYPROPS) {
+                // we won't collect orphan nodes in this mode, only empty properties would be collected.
+                return false;
+            }
+
```

Comment:
is this still needed? in theory it shouldn't ever call this method in that mode

Reply:
Sorry, missed to remove this.

Reply:
addressed in https://github.com/apache/jackrabbit-oak/pull/1639/commits/59076e6674847f4034add2144c3ca8bff31ac37b